docs/content/sips/012-signing-spin-releases.md

```diff
@@ -0,0 +1,206 @@
+title = "SIP 012 - "Signing the Spin releases"
+template = "main"
+date = "2023-03-01T01:01:01Z"
+
+---
+
+Summary: This improvement proposal describes the reasoning and implementation
+for signing Spin releases.
+
+Owners: radu@fermyon.com
+
+Created: March 2, 2023
+
+## Background
+
+Signing release artifacts for software offers a way for consumers of the software
+to verify the integrity of the package they downloaded. The signature should
+offer two guarantees: the author of the artifact is indeed the one expected by
+the consumer, and the content of the artifact has not been tampered with since
+its creation.
```

Comment:
I don't have any comment on the technology, which I realise is the focus of the SIP, but I'd find it really useful to understand what guarantees it provides, what attacks it mitigates, what malicious actions it allows a consumer to detect.  The proposal mentions "validate the integrity" but clearly means something more than file integrity (which is presumably guaranteed with the digest).  Is it about the attack of someone posting a false file somewhere and claiming it to be Spin?

Reply:
Besides the file integrity (which is now verifiable using a transparency log, rather than just using a file), a user can validate that the build was actually generated by the `fermyon/spin` repo, through a GitHub action workflow they can inspect at a given SHA or tag, and that us or someone else has not replaced the binary since the GitHub infrastructure built it at the time of the SHA or tag.


Besides this guarantee, this can be later expanded to include even more attestations (such as [SLSA](https://slsa.dev/provenance/v0.2), which could include even more provenance data).